@ecomplus/storefront-app/src/components/js/CreditCardForm.js

```diff
@@ -270,7 +275,10 @@ export default {
       alert.bin = !this.isNumberPotentiallyValid
       alert.date = !this.validateDate()
       alert.cvv = !this.validateCvv()
-      if (!alert.bin && !alert.date && !alert.cvv) {
+      if (!this.canHideHolderFields) {
+        alert.doc = !this.validateDoc()
+      }
+      if ((this.canHideHolderFields && !alert.bin && !alert.date && !alert.cvv) || (!this.canHideHolderFields && !alert.bin && !alert.date && !alert.cvv && !alert.doc)) {
```

Comment:
Essa linha aqui tá doideira em kk

Reply:
Se a pessoa coloca apenas a data de validade e o cvv, o submit não impede se o cpf não estiver preenchido. A ideia foi, se o holder name for igual, segue o jogo, não precisa ter outras checagens, apenas cvv, data de validade e o bin, se holder name e name for diferente, preciso checar se cpf foi preenchido.

Reply:
> o submit não impede se o cpf não estiver preenchido

Não impede porque os campos não eram requeridos, natural, se for só pra isso bastava adicionar o campo como requerido, não precisava de todo o resto do JS adicional...

Reply:
No exemplo que eu mandei o checkout era bloqueado:

![Screenshot 2023-09-20 at 23-59-28 My Shop](https://github.com/ecomplus/storefront/assets/10326572/b79724b9-34fe-4b42-b7c2-d476d3885330)

Reply:
Também não passa com o campo vazio ele estando requerido

Reply:
O que eu fiz com o `alert.cvv` por exemplo é porque não tinha um `<input-cvv>` que validava a string, no caso do doc number já existe, não tem porque duplicar a validação